src/clients/c++/request_grpc.cc

```diff
@@ -333,58 +341,79 @@ SharedMemoryControlGrpcContextImpl::RegisterSharedMemory(
     const std::string& name, const std::string& shm_key, const size_t offset,
     const size_t byte_size)
 {
-  return SendRequest(
-      name, SharedMemoryControlRequest::REGISTER, shm_key, offset, byte_size);
+  return SendRequest(name, REGISTER, shm_key, offset, byte_size, nullptr);
 }
 
 Error
 SharedMemoryControlGrpcContextImpl::UnregisterSharedMemory(
     const std::string& name)
 {
-  return SendRequest(name, SharedMemoryControlRequest::UNREGISTER, "", 0, 0);
+  return SendRequest(name, UNREGISTER, "", 0, 0, nullptr);
 }
 
 Error
 SharedMemoryControlGrpcContextImpl::UnregisterAllSharedMemory()
 {
-  return SendRequest("", SharedMemoryControlRequest::UNREGISTER_ALL, "", 0, 0);
+  return SendRequest("", UNREGISTER_ALL, "", 0, 0, nullptr);
 }
 
 Error
 SharedMemoryControlGrpcContextImpl::SendRequest(
```

Comment:
Why use a common routine? There is not that much common code across the request types and it just makes it confusing.

Reply:
Will split this into different components and remove the SendRequest function